GitUI/UserControls/InteractiveGitActionControl.cs

```diff
@@ -0,0 +1,241 @@
+
+            if (Module.InTheMiddleOfRebase())
+            {
+                SetGitAction(GitAction.Rebase, hasConflicts);
+                return;
+            }
+
+            if (Module.InTheMiddleOfMerge())
+            {
+                SetGitAction(GitAction.Merge, hasConflicts);
+                return;
+            }
+
+            if (Module.InTheMiddleOfPatch())
+            {
+                SetGitAction(GitAction.Patch, hasConflicts);
+                return;
+            }
+
+            SetGitAction(GitAction.None, false);
+        }
+
+        private void SetGitAction(GitAction action, bool hasConflicts)
+        {
+            if ((action == _action) && (hasConflicts == _hasConflicts))
+            {
+                // nothing to do
+                return;
+            }
+
+            _action = action;
+            _hasConflicts = hasConflicts;
+
+            // remove old controls
+            ButtonContainer.Controls.Clear();
+
+            if (_action == GitAction.None)
+            {
+                Visible = false;
+                return;
+            }
+
+            IconLabel.Image = _hasConflicts ? global::GitUI.Properties.Images.SolveMerge : global::GitUI.Properties.Resources.information;
+            BackColor = _hasConflicts ? System.Drawing.Color.Orange : System.Drawing.Color.LightSkyBlue;
+
+            string actionStr = "";
+
+            switch (_action)
+            {
+                case GitAction.Bisect:
+                    actionStr = _bisect.Text;
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+                case GitAction.Rebase:
+                    actionStr = _rebase.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+                case GitAction.Merge:
+                    actionStr = _merge.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    break;
+                case GitAction.Patch:
+                    actionStr = _patch.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+            }
+
+            TextLabel.Text = string.Format(
+                _hasConflicts ?
+                    _progressWithConflictsMessage.Text :
+                    _progressMessage.Text,
+                actionStr);
+
+            Visible = true;
+        }
+
+        private void ResolveButton_Click(object sender, EventArgs e)
+        {
+            GitModuleForm form = FindForm() as GitModuleForm;
```

Comment:
I suggest to create a private property for this as it is used in few places.

Reply:
I've created a private property, but not sure if I'm using it correctly. Please advise if more changes are needed on this.